app/services/facebook/send_on_facebook_service.rb

```diff
@@ -20,15 +20,30 @@ def perform_reply
   end
 
   def send_message_to_facebook(delivery_params)
-    result = Facebook::Messenger::Bot.deliver(delivery_params, page_id: channel.page_id)
-    parsed_result = JSON.parse(result)
+    parsed_result = deliver_message(delivery_params)
+    return if parsed_result.nil?
+
     if parsed_result['error'].present?
       message.update!(status: :failed, external_error: external_error(parsed_result))
       Rails.logger.info "Facebook::SendOnFacebookService: Error sending message to Facebook : Page - #{channel.page_id} : #{result}"
     end
+
     message.update!(source_id: parsed_result['message_id']) if parsed_result['message_id'].present?
   end
 
+  def deliver_message(delivery_params)
+    result = Facebook::Messenger::Bot.deliver(delivery_params, page_id: channel.page_id)
+    JSON.parse(result)
+  rescue JSON::ParserError
+    message.update!(status: :failed, external_error: 'Facebook was unable to process this request')
+    Rails.logger.error "Facebook::SendOnFacebookService: Error parsing JSON response from Facebook : Page - #{channel.page_id} : #{result}"
```

Comment:
```suggestion
    Rails.logger.error "Facebook::SendOnFacebookService: Error parsing JSON response from Facebook : Page - #{channel.page_id} : #{result.inspect}"
```

Reply:
Not sure if we need to use inspect, it'll be converted to it's string representation anyway.